doc/Makefile

```diff
@@ -104,16 +101,18 @@ htmlhelp:
 	@echo "Build finished; now you can run HTML Help Workshop with the" \
 	      ".hhp project file in $(BUILDDIR)/htmlhelp."
 
-latex:
+latex: logo
 	mkdir -p $(BUILDDIR)/latex $(BUILDDIR)/doctrees
 	$(SPHINXBUILD) -b latex $(ALLSPHINXOPTSlatex) $(BUILDDIR)/latex
 	@echo
 	@echo "Build finished; the LaTeX files are in $(BUILDDIR)/latex."
-	@echo "Set the environment variable LATEXMKOPTS='-xelatex -silent'"
-	@echo "And run \`make all' in that directory to run these through xelatex."
+	@echo "Run \`make all' in that directory to run these through xelatex."
 
 pdf pdflatex latexpdf: latex
-	$(MAKE) -C $(BUILDDIR)/latex
+	@$(MAKE) -C $(BUILDDIR)/latex all-pdf \
+	LATEXOPTS="-halt-on-error -interaction=batchmode" \
+	LATEXMKOPTS="-xelatex" \
+	|| echo "LaTeX error"'!'" Check .log in $(BUILDDIR)/latex for lineno in .tex file."
```

Comment:
This is the correct way, now that the Sphinx Makefile and latexmkrc are correctly configured, to use LATEXOPTS and LATEXMKOPTS. The `-silent` is too drastic for Latexmk in case of problem, or for example when Latexmk does many successive passes, the reasons get lost. So we pass batchmode to the xelatex excutable rather to keep avoiding humongous console output.